ompi/mca/osc/ucx/osc_ucx_passive_target.c

```diff
@@ -158,9 +158,22 @@ int ompi_osc_ucx_unlock(int target, struct ompi_win_t *win) {
     opal_hash_table_remove_value_uint32(&module->outstanding_locks,
                                         (uint32_t)target);
 
-    ret = opal_common_ucx_wpmem_flush(module->mem, OPAL_COMMON_UCX_SCOPE_EP, target);
-    if (ret != OMPI_SUCCESS) {
-        return ret;
+    if (module->flavor == MPI_WIN_FLAVOR_DYNAMIC) {
+        for (uint64_t i = 0; i < module->state.dynamic_win_count; i++) {
+            ret = opal_common_ucx_wpmem_flush(module->local_dynamic_win_info[i].mem , OPAL_COMMON_UCX_SCOPE_EP, target);
+            if (ret != OMPI_SUCCESS) {
+                return ret;
+            }
+        }
+        ret = opal_common_ucx_wpmem_flush(module->dynamic_mem, OPAL_COMMON_UCX_SCOPE_EP, target);
```

Comment:
What is `dynamic_mem` used for? Do we ever issue operations on it?

Reply:
Yes, dynamic_mem is a special kind of wpmem whose rkeys/addrs keep changing during the application runtime. In dynamic windows, when users provide a remotely accessible address, we need to read the rkey info from the remote process and update dynamic_mem based on that.